packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,306 @@
+
+-   `key`: string (property type matches component type, e.g. `number` for number mapping)
+-   `description`: string
+
+_Content_
+
+-   exactly one `template` element, containg some combination of text, Nimble icons, or `nimble-spinner`
+
+### Anatomy
+
+#### `nimble-table-column-mapping`
+
+```HTML
+<template slot="${x => x.columnInternals.uniqueId}">
+    <slot
+        ${slotted('mappings')}
+        name="mapping">
+    </slot>
+    <span class="header-content">
+        <slot></slot>
+    </span>
+</template>
+```
+
+Cell view:
+
+```HTML
+${repeat(x => (x.column as TableColumnMapping).mappings,
+    html<TableColumnMapping, TableColumnMappingCellView>`
+        ${when((x, c) => x.key === (c.parent as TableColumnMappingCellView).cellRecord.value,
+            html<TableColumnMapping>`
+                ${repeat(x => x.mappedTemplate,
+                    html<HTMLTemplateElement>`${x => new ViewTemplate(x.innerHTML, [])}`
+                )}
+            `
+        )}
+    `
+)}
+```
+
+Header view:
+
+Note the following requires that `TableColumnMappingHeaderView` has a reference to the column with which it is associated. This is needed to enumerate the column's mapping elements.
+
+```HTML
+${repeat(x => (x.column as TableColumnMapping).mappings,
+    html<TableColumnMapping, TableColumnMappingHeaderView>`
+        ${when((x, c) => x.key === (c.parent as TableColumnMappingHeaderView).groupHeaderValue,
+            html<TableColumnMapping>`
+                ${repeat(x => x.mappedTemplate,
+                    html<HTMLTemplateElement>`${x => new ViewTemplate(x.innerHTML, [])}`
+                )}
+            `
+        )}
+    `
+)}
+```
+
+For each mapping, if the key matches the value for this row, get the mapping's contained `template` and render its `innerHTML`.
+
+#### `nimble-mapping-string/number/boolean`
+
+```HTML
+<template slot="mapping">
+    <slot
+        ${slotted({ property: 'mappedTemplate', filter: x => x instanceof HTMLTemplateElement })}
+    >
+    </slot>
+</template>
+```
+
+### Grouping
+
+Will support grouping by the record value. The grouping header will display the rendered html mapped to that value. If multiple different values map to the same rendered html (e.g. icon), there may be separate groups that visually look like they should be combined into one group. We could instead render the record value into the header, but that value is not otherwise exposed to the user. It may not even have much meaning (e.g. some numeric code). I think it is more reasonable to display the rendered html.
+
+### Sorting
+
+Sorting has the same issue as grouping, where visually identical cells could end up not sorted contiguously. Again, I believe this is reasonable.
+
+### Sizing
+
+We will support fixed or fractional widths for this column type.
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created.
```

Comment:
I'm guessing directives will be created for the mapping elements as well? It would be good to be explicit about that here.

Reply:
Made that explicit.